src/backend/distributed/utils/node_metadata.c

```diff
@@ -106,7 +107,16 @@ master_remove_node(PG_FUNCTION_ARGS)
 	EnsureSuperUser();
 
 	hasShardPlacements = NodeHasActiveShardPlacements(nodeNameString, nodePort);
-	if (hasShardPlacements)
+	if (hasShardPlacements && force)
```

Comment:
I think this would be cleaner like:

```c
if (hasShardPlacements)
{
    if (!force)
    {
        ereport(ERROR, /* ... */);
    }

    ereport(NOTICE, /* ... */);
}
```